### ISSUE TITLE
Infinite fsck loop in emergency mode with separate /var that requires manual fsck

### ISSUE THREAD

**Issue by ihenry1** [systemd/systemd#9171]
**systemd version the issue has been seen with**
> 238.76

**Used distribution**
 > Arch Linux

**Expected behaviour you didn't see**
 > systemd stops trying to fsck /var after it fails and the system goes into emergency mode

**Unexpected behaviour you saw**
 > systemd keeps trying to re-run fsck on /var forever, even after entering emergency mode. This means I'm not able to run fsck manually to fix the problem with the filesystem.

```
# systemctl | grep start
var.mount                                                                                 loaded inactive   dead      start /var                                                                         
systemd-fsck@dev-disk-by\x2duuid-843e72f6\x2d7421\x2d4e30\x2d9900\x2dcd8baac1d8fc.service loaded activating start     start File System Check on /dev/disk/by-uuid/843e72f6-7421-4e30-9900-cd8baac1d8fc  
systemd-rfkill.service                                                                    loaded inactive   dead      start Load/Save RF Kill Switch Status
# systemctl list-jobs
JOB UNIT                                                           TYPE  STATE  
749 var.mount                                                      start waiting
751 systemd-fsck@dev-dis…1\x2d4e30\x2d9900\x2dcd8baac1d8fc.service start running
746 systemd-rfkill.service                                         start waiting

3 jobs listed.
```

Trying to cancel these start jobs with ```systemctl cancel 746 751 749``` doesn't work, the jobs are immediately restarted. ```systemctl stop``` on any of those services says that a job was cancelled and does nothing.

It's systemd-rfkill.service that's pulling in var.mount, I don't know what keeps restarting that. There's a systemd-rfkill.socket as well, so maybe it's getting socket-activated by something?

```
# systemctl list-dependencies systemd-rfkill
systemd-rfkill.service
● ├─-.mount
● ├─sys-devices-virtual-misc-rfkill.device
● ├─system.slice
● └─var.mount
```

I worked around this by stopping systemd-fsck just before it exited, then running fsck manually, then letting it continue:

```
<type root password>
# pkill -STOP systemd-fsck    # blindly type this under systemd-fsck's progress output
# gdb -p $(pgrep systemd-fsck)
(gdb) break exit
(gdb) signal SIGCONT
(gdb) c
<hit breakpoint on exit>
(gdb) shell
# fsck /dev/disk/by-uuid/843e72f6-7421-4e30-9900-cd8baac1d8fc
# exit
(gdb) q
```

Temporarily masking systemd-fsck@.service, var.mount or systemd-rfkill.service might work too, I didn't try that.

**Steps to reproduce the problem**
1. Have a separate partition for /var.
2. Break the filesystem on /var in a way that requires running fsck manually to fix, or I think any non-zero exit from fsck will work - do something will make systemd-fsck exit with failure for /var.
3. Reboot.

**Comment by ihenry1** [comment]
Journal for that boot with systemd.log_level=debug, up to when I put the real systemd-fsck back and exited emergency mode, at https://gist.github.com/ihenry1/3457cb5c9ccf8722600512538de5ce89, first "fsck" failure at [line 3680](https://gist.github.com/ihenry1/3457cb5c9ccf8722600512538de5ce89#file-journalctl_-b_-u_21-02-07-log-L3680), one loop at [lines 4382-4415](https://gist.github.com/ihenry1/3457cb5c9ccf8722600512538de5ce89#file-journalctl_-b_-u_21-02-07-log-L4382-L4415)

This is with systemd-fsck replaced with this script to simulate fsck saying that it needs to be run manually.
```bash
#!/bin/sh
echo "systemd-fsck stub: $@" > /dev/console
sleep 10    # avoid getting rate limited, real fsck didn't
if [ "$1" = /dev/disk/by-uuid/843e72f6-7421-4e30-9900-cd8baac1d8fc ]; then
    exit 4  # Filesystem errors left uncorrected
fi
exit 0
```

**Comment by poettering** [comment]
ah, this is triggered by rfkill... i figure we should not subscribe to rfkill events until we know /var is accessible and we can safely access it. Otherwise the rfkill service will be triggered due an rfkill event, but then needs to wait for /var to work.

**Comment by poettering** [comment]
Fix waiting in #10904
